core/store/src/trie/split_state.rs

```diff
@@ -150,9 +150,13 @@ impl ShardTries {
         let mut store_update = self.store_update();
         for (shard_uid, changes) in changes_by_shard {
             // Here we assume that state_roots contains shard_uid, the caller of this method will guarantee that.
-            let trie_changes =
-                self.get_trie_for_shard(shard_uid, state_roots[&shard_uid]).update(changes)?;
+            let trie_changes = self
+                .get_trie_for_shard(shard_uid, state_roots[&shard_uid])
+                .update(changes.clone())?;
```

Comment:
Is there any chance to use references here? So that we don't need to clone.

Reply:
Removed in current PR but will keep in mind to try to not clone here.